dialog_design.md

```diff
@@ -0,0 +1,68 @@
+# Dialogs
+
+Goal: Display a message and an optional graphic to all users, or some, or one, and return a response.
+
+## Architecture
+
+Create a new `Dialog` class that manages calls and responses. It is also a good place to park static convenience
+functions and event listeners. I propose a couple different setup options:
+
+1. `type` - What sort of response we should expect.
+	1. A `message` dialog presents an OK button, and returns no value, just a confirmation that OK was clicked.
+	2. A `confirm` dialog presents OK and Cancel buttons, and returns a boolean.
```

Comment:
I like this. Perhaps an array of buttons: ['ok', 'cancel'].

Reply:
Yeah, if we make this an array of buttons, 'message' and 'confirm' are really the same thing. then the type field really comes down to 'add a text field'. 
We would need to also add a "default result" for if you just press enter on a text field

---
In reply to: [320502157](https://github.com/Microsoft/mixed-reality-extension-sdk/pull/444#discussion_r320502157) [](ancestors = 320502157)

Reply:
That's not bad. Instead of static ok/cancel, take an array of strings and we'll have that many buttons. It would respond with the text string on the clicked button.

---
In reply to: [321510661](https://github.com/microsoft/mixed-reality-extension-sdk/pull/444#discussion_r321510661) [](ancestors = 321510661,320502157)